src/Modules/OrchardCore.Commerce/Services/PriceService.cs

```diff
@@ -17,7 +17,11 @@ public class PriceService : IPriceService
 
     public async Task<IList<ShoppingCartItem>> AddPricesAsync(IList<ShoppingCartItem> items)
     {
-        foreach (var priceProvider in _providers.OrderBy(provider => provider.Order))
+        var providers = await _providers
+            .OrderBy(provider => provider.Order)
+            .WhereAsync(provider => provider.IsApplicableAsync(items));
```

Comment:
You forgot to do the same for `AddPriceAsync` here:
https://github.com/OrchardCMS/OrchardCore.Commerce/blob/b671bae20827a73cb288d571d5d4fbc17f82fc7f/src/Modules/OrchardCore.Commerce/Services/PriceService.cs#L35

Also can you explain why we needed separate `AddPriceAsync` method at all? Seems to me that it should be just an extension method for `IPriceService` that returns something like `(await priceService.AddPricesAsync(new [] { item })).Single()` instead of a full implementation. Then we wouldn't need `IPriceProvider.AddPriceAsync()` either.

Reply:
I added this back then when I did the Stripe implementation. I will try your suggestion.